spotinst/commons/common_set_credential_aws.go

```diff
@@ -0,0 +1,88 @@
+package commons
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/spotinst/spotinst-sdk-go/service/account/providers/aws"
+	"log"
+)
+
+const (
+	SetCredentialAWSResourceName ResourceName = "spotinst_set_credential_aws"
+)
+
+var CredentialAWSResource *CredentialAWSTerraformResource
+
+type CredentialAWSTerraformResource struct {
+	GenericResource
+}
+
+type AWSCredentialWrapper struct {
+	credential *aws.Credential
+}
+
+func NewCredentialAWSResource(fieldsMap map[FieldName]*GenericField) *CredentialAWSTerraformResource {
+	return &CredentialAWSTerraformResource{
+		GenericResource: GenericResource{
+			resourceName: SetCredentialAWSResourceName,
+			fields:       NewGenericFields(fieldsMap),
+		},
+	}
+}
+
+func (res *CredentialAWSTerraformResource) OnCreate(
+	resourceData *schema.ResourceData,
+	meta interface{}) (*aws.Credential, error) {
+
+	if res.fields == nil || res.fields.fieldsMap == nil || len(res.fields.fieldsMap) == 0 {
+		return nil, fmt.Errorf("resource fields are nil or empty, cannot create")
+	}
+
+	credentialWrapper := NewCredentialWrapper()
+
+	for _, field := range res.fields.fieldsMap {
+		if field.onCreate == nil {
+			continue
+		}
+		log.Printf(string(ResourceFieldOnCreate), field.resourceAffinity, field.fieldNameStr)
+		if err := field.onCreate(credentialWrapper, resourceData, meta); err != nil {
+			return nil, err
+		}
+	}
+	return credentialWrapper.GetCredential(), nil
+}
+
```

Comment:
let's follow same order in all resources: OnCreate, OnRead, NewWrapper, get and set methods

Reply:
Arranged in order OnCreate, OnRead, NewWrapper, get and set methods.